### PR TITLE
CI and Profile Support

### PR DESCRIPTION
**Pull request by ben-c-at-moz**
# Description

* Does not write test failure artifacts unless --ci is set
* Sets Mac and Win Fx to be the correct Beta versions
* Adds a test to grab the Fx version in `meta`
* Adds a fixture in conftest to prep to use profiles in tests (I did check that it works)
* Adds first profile to test with, called `theme_change`.

# Comments / Concerns

Fx and gecko locations are now hardcoded, this should be addressed in a future PR (which should also deprecate `collect_executables.sh`).

### CODE REVIEW

**Comment by ben-c-at-moz** [comment]
> Looks great and was able to run a test using the fixture with theme_change profile.
> 
> However, there is still the 'beta' folder with cruft in it that should be removed.

That folder shouldn't be there anymore, it's not in the changeset. Try removing it from your local and repulling (with my branch checked out)

File: tests/bookmarks_and_history/test_bookmark_website_via_star_button.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ def test_bookmark_website_via_star(driver: Firefox):
     # instantiate object
     nav = Navigation(driver)
     panel = PanelUi(driver)
+    # random comment
```
Comment (sarinali) [inline tests/bookmarks_and_history/test_bookmark_website_via_star_button.py:16]: meant to be here?

File: profiles/theme_change.zip
Comment (sarinali) [inline profiles/theme_change.zip:1]: so im assuming this is unzipped and used right? sorry just want to ask about how it works first

**Review by sarinali — APPROVED** [review]
lgtm, can you just answer my question?

File: tests/bookmarks_and_history/test_bookmark_website_via_star_button.py (hunk 1 above)
Comment (ben-c-at-moz) [reply tests/bookmarks_and_history/test_bookmark_website_via_star_button.py:16]: LOOOOL, the change that started it all. No, that should be removed.

File: profiles/theme_change.zip
Comment (ben-c-at-moz) [reply profiles/theme_change.zip:1]: Yep, check out the driver fixture in Conftest.

**Pull request merged**